### DISCUSSION TITLE
one think about mctp packet sending

### DISCUSSION THREAD

**Discussion by maxwell652** [openbmc/libmctp#14] [General]
hello guys, I have read the source code of the [mctp_message_tx_on_bus](https://github.com/openbmc/libmctp/blob/b3de343e8b8952d337102df130a02dd13cce5b83/core.c#L768) function and learned that the current implementation is based on BTU splitting the original msg into multiple mctp packets for sending. But currently, we first allocate memory for all MCTP packets and then send them one by one. When a certain MCP packet fails to send, it will continue to send the next packet.

Can we change these implementations to only allocate memory only for one MCTP packet to be sent. And if the transmission is successful, continue to allocate memory and send for the next packet. If the sending fails, terminate the entire sending process.

In this way, it will save some memory resources, and if the original msg is particularly large (such as exceeding 4kB), the saved resources will be considerable. Because some embedded systems have very limited memory resources.

**Comment by amboar** [comment]
I think the best way to move forward with this is for you to to send patches implementing the behaviour you need. That will give us something concrete to evaluate.

Currently libmctp takes patches through the OpenBMC Gerrit instance:

https://gerrit.openbmc.org/

**Reply by maxwell652** [reply]
ok, I'll try it later.
